uplink_python/access.py

```diff
@@ -80,9 +81,18 @@ def derive_encryption_key(self, passphrase: str, salt: str):
         #
         # if error occurred
         if bool(encryption_key_result.error):
-            raise _storj_exception(encryption_key_result.error.contents.code,
-                                   encryption_key_result.error.contents.message.decode("utf-8"))
-        return encryption_key_result.encryption_key
+            error_code = encryption_key_result.error.contents.code
```

Comment:
It looks like this is a pretty common pattern here. Also, though, it looks kind of fragile. Even if we are very careful this time, the next person to work on this code could easily mess something up.

It might be worth isolating this pattern to a few utility functions. Something like:

```python
def unwrap_libuplink_result(result_object, finalizer, attribute_name):
    if bool(result_object.error):
        error_code = result_object.error.contents.code
        error_msg = result_object.error.contents.message.decode("utf-8")
        finalizer(result_object)
        raise _storj_exception(error_code, error_msg)
    result = getattr(result_object, attribute_name)
    finalizer(result_object)
    return result

def unwrap_encryption_key_result(result_object, uplink_handle):
    return unwrap_libuplink_result(result_object, uplink_handle.m_libuplink.uplink_free_encryption_key_result, 'encryption_key')

def unwrap_project_result(result_object, uplink_handle):
    return unwrap_libuplink_result(result_object, uplink_handle.m_libuplink.uplink_free_project_result, 'project')

# (and similar methods for StringResult, AccessResult, etc)
```

Then in this function you could replace this whole ending (everything from line 81 on) with:

```python
    return unwrap_encryption_key_result(encryption_key_result, self.uplink)
```

The preexisting code in this repo is still pretty messy, though, so that could only help so much.